src/main/java/io/cryostat/agent/MainModule.java

```diff
@@ -236,10 +236,20 @@ public static SSLContext provideClientSslContext(
             @Named(ConfigModule.CRYOSTAT_AGENT_WEBCLIENT_TLS_CLIENT_AUTH_KEY_PASS_CHARSET)
                     String clientAuthKeyPassFileCharset,
             @Named(ConfigModule.CRYOSTAT_AGENT_WEBCLIENT_TLS_CLIENT_AUTH_KEY_MANAGER_TYPE)
-                    String clientAuthKeyManagerType) {
+                    String clientAuthKeyManagerType,
+            @Named(ConfigModule.CRYOSTAT_AGENT_BASEURI) URI baseUri,
+            @Named(ConfigModule.CRYOSTAT_AGENT_WEBCLIENT_TLS_REQUIRED) boolean tlsEnabled) {
         try {
             KeyManager[] keyManagers = null;
-            if (clientAuthCertPath.isPresent() && clientAuthKeyPath.isPresent()) {
+            if (clientAuthCertPath.isPresent() && clientAuthKeyPath.isPresent() && tlsEnabled) {
```

Comment:
I think the `tlsEnabled` check should be separated out. The variable/parameter should probably also be renamed `tlsRequired` to match the configuration property name, for clarity of reading.

The `clientAuthCertPath` and `clientAuthKeyPath` are there to enable configuration of the Agent's HTTP Client instance that it uses for talking to the Cryostat server. So I think the check we want is like:

```
if (tlsRequired) {
  if (!baseUri.getScheme().equals("https")) throw new IllegalArgumentException("Base URI must be https");
}
```

This way, if the Agent has TLS enforcement turned on, and the user configures it with a non-HTTPS server base URI, then the Agent will refuse to make that connection. If the user has configured with with an HTTPS server base URI, but did not configure the client auth cert/key paths, then the connection should already fail at a later point because the TLS auth will fail.